sentry-core/src/main/java/io/sentry/core/ShutdownHookIntegration.java

```diff
@@ -0,0 +1,21 @@
+package io.sentry.core;
+
+import io.sentry.core.util.Objects;
+import org.jetbrains.annotations.NotNull;
+
+/** Registers hook that closes {@link Hub} when main thread shuts down. */
+public final class ShutdownHookIntegration implements Integration {
+
+  private final @NotNull Runtime runtime;
+
+  public ShutdownHookIntegration(final @NotNull Runtime runtime) {
+    this.runtime = Objects.requireNonNull(runtime, "Runtime is required");
+  }
```

Comment:
```suggestion
  ShutdownHookIntegration(final @NotNull Runtime runtime) {
    this.runtime = Objects.requireNonNull(runtime, "Runtime is required");
  }
  
  public ShutdownHookIntegration() {
    this(Runtime.getRuntime());
  }
```

Reply:
... and invoke non-param constructor in `SentryOptions`, right?

Reply:
yep, this is just for the sake of testing, one only needs `new ShutdownHookIntegration()`